DomainContent/Rust/DJ-Tools/DJ_Input_Zone_Client.js

```diff
@@ -0,0 +1,382 @@
+            }
+            this.endPoints.forEach(function(endPoint) {
+                Entities.callEntityServerMethod(
+                    endPoint.id, 
+                    'edit', 
+                    [stringifiedRange, stringifiedDirections]
+                );
+            });
+        }
+    };
+
+    function EndPoint(name, id, position, endPointGroupID) {
+        this.name = name;
+        this.id = id;
+        this.position = position;    
+        this.endPointGroupID = endPointGroupID;        
+    }
+
+    function Debug(name, id) {
+        this.name = name;
+        this.id = id;
+        this.position = Entities.getEntityProperties(this.id, ['position']).position;
+        this.overlay = Overlays.addOverlay("text3d", {
+            position: Vec3.sum(this.position, vec(0,1,0)),
+            parentID: this.id,
+            isFacingAvatar: true,
+            lineHeight: 0.09,
+            text: "test"
+        });
+    }
+
+    Debug.prototype = {
+        getPosition: function() {
+            this.position = Entities.getEntityProperties(this.id, ['position']).position;
+            return this.position;
+        },
+        updateOverlay: function() {
+            var text = JSON.stringify(debugInfo).split(",").join("\n");
+            var props = {
+                text: text
+            };
+            Overlays.editOverlay(this.overlay, props);
+        }
+    };
+
+    // Procedural Functions
+    function scanForInputs() {
+        var matchedEnts = [];
+        var searchRadius = largestAxisVec(currentProps.dimensions);
+        var entList = Entities.findEntities(currentProps.position,searchRadius);
+        entList.forEach(function(ent) {
+            var userData = gProps(ent, ["userData"]).userData;
+            try {
+                userData = JSON.parse(userData);
+                if (userData.performance) {
+                    if (userData.performance.type === INPUT) {
+                        var inputProps = gProps(ent);
+                        var name = inputProps.name;
+                        var id = inputProps.id;
+                        var position = inputProps.position;
+                        var dimensions = inputProps.dimensions;
+                        var endPointGroups = userData.performance.endPointGroups; 
+                        var directionArray = userData.performance.directionArray; 
+                        matchedEnts.push(
+                            new Input(
+                                name, 
+                                id, 
+                                position, 
+                                dimensions, 
+                                directionArray, 
+                                endPointGroups
+                            )
+                        );
+                    }
+                }
+            } catch (e) {
+                console.log(e);
+            }
+        });
+        return matchedEnts;
+    }
+
+    function scanForEndpoints() {
```

Comment:
Can these be consolidated into a single scan function that takes a parameter for what to search for?